### DISCUSSION TITLE
Analog temperature in NSPanel

### DISCUSSION THREAD

**Discussion by gonzzzales** [arendst/Tasmota#15737] [Q&A]
Hi all!
I bought two devices NSPanel by Sonoff and flashed to Tasmota. Unfortunately both devices show the wrong analog temperature
The DSB20 sensor shows temperature correctly
`12:37:51.894 RSL: SENSOR = {"Time":"2022-06-03T12:37:51","ANALOG":{"Temperature1":5.2},"DS18B20":{"Id":"C8E0AABDCCFF","Temperature":25.0},"ESP32":{"Temperature":53.9},"TempUnit":"C"}`

`12:43:53.020 RSL: SENSOR = {"Time":"2022-06-03T12:43:52","ANALOG":{"Temperature1":5.4},"DS18B20":{"Id":"FAE0AB6D6DFF","Temperature":28.0},"ESP32":{"Temperature":37.8},"TempUnit":"C"}`

Is it a malfunction of the devices or did I set them up incorrectly? resistance of the NTC sensor is 8.8kOm.

i use this template {"NAME":"NSPanel","GPIO":[0,0,0,0,3872,0,0,0,0,0,32,0,0,0,0,225,0,480,224,1,0,0,0,33,0,0,0,0,0,0,0,0,0,0,4736,0],"FLAG":0,"BASE":1,"CMND":"ADCParam 2,11200,10000,3950 | Sleep 0 | BuzzerPWM 1"}

**Comment by atirage** [comment]
Hi, 

I've noticed the same issue with my panel. It seems to me that the template ADC parameters are not quite correct for this particular sensor.
This is what the Tasmota wiki says about it (https://tasmota.github.io/docs/ADC/):

>[ADC] analog input tuning parameters.
 On ESP32 x is channel 1..8

 >sensor, param1, param2, param3, param4

 >sensor values:

 >    2 = Temperature [Steinhart-Hart thermistor equation](https://en.wikipedia.org/wiki/Steinhart%E2%80%93Hart_equation) parameters:

>param1 = NTC Voltage bridge resistor in Ohms (default = 32000)

>param2 = NTC Resistance in Ohms (default = 10000)

>param3 = NTC Beta Coefficient (default = 3350)

As you state, the NTC is 8.8kOhm, the template uses 10k.
Anyways, I've added the following command in the Tasmota console and at least in the low twenties the temperature reading became much more accurate:
`ADCParam1 2,12400,8800,3950`

Feel free to try it out.

**Reply by nhatttr** [reply]
It's work

**Reply by matsch-hasi** [reply]
it is also on my side working. Thx a lot

**Reply by txtweet** [reply]
It's a great fix, thank you!

**Reply by Arek2PL** [reply]
Thank You!

**Comment by Maxickode** [comment]
Hi,

i have the same problem with my new NSPanel. Flashed it today with Tasmota 15.1.0.1 and got around -4 degrees Celsius.
I changed the parameters like suggested and now i am around 15°. It should be around 22°.

I found the "GPIO Viewer" within the Tools and there i saw that GPIO38 shows a number which is close to the right temp.
<img width="184" height="120" alt="grafik" src="https://github.com/user-attachments/assets/c3bb151b-c037-4708-af6c-ebb22312ecbe" />

I hope someone has a clue, why it still shows the wrong temp in the menu, but on the GPIO it looks pretty good.
I searched for something else than "ADC Temp" already, but couldn't anything which fits also for temperature.
<img width="446" height="66" alt="grafik" src="https://github.com/user-attachments/assets/b86f8a67-2d08-443e-a492-75310e717dbb" />

KR,
Max

**Reply by sfromis** [reply]
The number you see with GPIO Viewer is millivolts, with variations I'd not expect any real similarity with temperature. The logic with resistance values is already described in this thread. Not at all straightforward, but can work out with detailed info about the hardware.
